packages/plugin-commands-outdated/src/outdated.ts

```diff
@@ -244,6 +251,31 @@ ${renderCurrent(outdatedPkg)} ${chalk.grey('=>')} ${renderLatest(outdatedPkg)}`
     .join('\n\n') + '\n'
 }
 
+function renderOutdatedJSON (outdatedPackages: readonly OutdatedPackage[], opts: { long?: boolean }) {
+  const outdatedPackagesJSON = {} as {
+    [outdatedPackageName: string]: {
+      current: string
+      latest: string
+      dependencyType: DependenciesField
+      details?: string
+    }
+  }
+  for (const outdatedPkg of sortOutdatedPackages(outdatedPackages)) {
+    const { packageName, belongsTo } = outdatedPkg
+    outdatedPackagesJSON[packageName] = {
+      current: renderCurrent(outdatedPkg),
+      latest: extractLatest(outdatedPkg),
```

Comment:
I don't think you need a function. Just return
```suggestion
      latestVersion: outdatedPkg.latestManifest?.version,
      latestDeprecated: outdatedPkg.latestManifest?.deprecated,
```

Reply:
same as https://github.com/pnpm/pnpm/pull/5582#discussion_r1014654608

Reply:
I thought user want to know which package is deprecated, but they only know which package is deprecated by checking `current` field is equal to `latestVersion` field and this is not friendly. So I added `deprecated` field.